models/articles-models.js

```diff
@@ -21,21 +27,33 @@ exports.selectArticles = () => {
      FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;`).then((result) => {
      
       
+      if(result.rows.length===0){
+        return Promise.reject({status:404, msg:"article doesn't exist"})
+      } 
+      else{
       return result.rows;
+      }
     });
   };
 
   exports.selectCommentsByArticle = (id) =>{
-    return db.query(` SELECT   comment_id,
+    if(isNaN(id)){return Promise.reject({status:400, msg:"Invalid id" })}
```

Comment:
Same thing, PSQL will send back an error if id is not a number, so you don't need to check for it here